### ISSUE TITLE
Embed interop types with internal accessibility

### ISSUE THREAD

**Issue by pharring** [dotnet/roslyn#183]
When embedding an interop assembly, the types (interfaces, typically) from that assembly become (or remain) public in the final assembly. This is, usually, not what you want. Particularly for private interops. Indeed, you can see this 'accident' in some Visual Studio assemblies such as Microsoft.VisualStudio.Shell.12.0.dll (look in the Microsoft.Internal.VisualStudio.Shell.Interop namespace)
While we can't change the default behavior without breaking compatibility, I propose a compiler option to change the accessibility of any embedded type to 'internal'. For maximum flexibility, I suppose this option could be applied optionally to each interop embedded. I don't think we need to extend it to the level of an individual type.
Suggestion:
Where today you specify an embedded assembly on the command-line via `/link:<file list>`, introduce a new option `/linkinternal:<file list>`

**Comment by pharring** [comment]
Actually, there might be a good argument for making the embedded types internal by default. The compiler is allowed to omit unused methods from embedded interfaces. So, exposing the interfaces publicly could lead to an incomplete definition.

**Comment by theoy** [comment]
I agree that internal by default is a better approach, but I think that this is unfortunately a compat break. We'll treat this a feature request to enable alternate behavior.

**Comment by pharring** [comment]
I was working on another fix with TypeIdentifiers and I discovered that the runtime refuses to load a type that is tagged with TypeIdentifier if it is not public. So perhaps this is a non-starter.
